File: src/main/java/plugin/Listeners.java
// This is free and unencumbered software released into the public domain.
// Author: NotAlexNoyle.
package plugin;

// Import libraries.
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;
import com.destroystokyo.paper.event.player.PlayerAdvancementCriterionGrantEvent;
import org.bukkit.GameMode;
// Hook into Bukkit's Listener.
public class Listeners implements Listener {
  @EventHandler(priority = EventPriority.HIGHEST)
	public void onAdvancementProgress(PlayerAdvancementCriterionGrantEvent event) {
		if(event.getPlayer().getGameMode() != GameMode.SURVIVAL) {
			event.setCancelled(true);
		}
	}
}

File: src/main/kotlin/plugin/Listeners.kt
// This is free and unencumbered software released into the public domain.
// Author: Sekalol15
package plugin

import org.bukkit.Material
import org.bukkit.entity.EntityType
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.entity.CreatureSpawnEvent
import org.bukkit.event.entity.EntityDeathEvent
import org.bukkit.inventory.ItemStack

import kotlin.random.Random

class Listeners : Listener {
    @EventHandler
    fun templaterEvent(event: CreatureSpawnEvent) {

        val TemplateInt = Config.getTemplateInt()
        val TemplateBool = Config.getTemplateBool()
        val TemplateDouble = Config.getTemplateDouble()

    }


}





